src/concrete/ml/pandas/_processing.py

```diff
@@ -86,9 +92,37 @@ def dequant(
     return x.astype(dtype)
 
 
-# Provide a way for users to pass string mappings
-# FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4342
-def pre_process_dtypes(pandas_dataframe: pandas.DataFrame) -> Tuple[pandas.DataFrame, Dict]:
+def check_schema_format(pandas_dataframe: pandas.DataFrame, schema: Optional[Dict] = None) -> None:
+    if schema is None:
+        return
+
+    if not isinstance(schema, dict):
+        raise ValueError(
+            "When set, parameter 'schema' must be a dictionary that associates some of the "
+            f"data-frame's column names to their value mappings. Got {type(schema)=}"
+        )
+
+    column_names = list(pandas_dataframe.columns)
+
+    for column_name, column_mapping in schema.items():
+        if column_name not in column_names:
+            # TODO: Is this check actually relevant ? Can't the schema provide more columns than the
+            # one found in the data-frame ?
+            raise ValueError(
```

Comment:
should we allow schema with column names that do not match the ones found in the given data-frame ?

Reply:
Imo raising the error as you are doing here is the correct behavior.